# WEB-API(Application Programming Interface)
- 웹서비스의 정보를 웹사이트를 통해 제공 : 크롤링으로 수집(비공식 수집)
- 웹API : AIP 정보로 수집(공식적으로 수집)

## Kakao API
1. 카카오 개발자 가입 : https://developers.kakao.com
2. 앱 ID 등록(개발자 계정)
3. REST API 사용 : https://developers.kakao.com/docs/restapi

In [31]:
import requests

In [32]:
address = '서울특별시 구로구 디지털로 306'
app_key = 'KakaoAK ' + '1b131358d811397813db74376a179722'
url = 'https://dapi.kakao.com/v2/local/search/address.json' # endpoint
params = {
    'query' : address,
    'page' : '',
    'AddressSize' : ''
}
headers = {'Authorization' : app_key}
response = requests.get(url, headers=headers, params=params)
response.status_code

200

In [18]:
response.json()

{'documents': [{'address': {'address_name': '서울 구로구 구로동 182-13',
    'b_code': '1153010200',
    'h_code': '1153054000',
    'main_address_no': '182',
    'mountain_yn': 'N',
    'region_1depth_name': '서울',
    'region_2depth_name': '구로구',
    'region_3depth_h_name': '구로3동',
    'region_3depth_name': '구로동',
    'sub_address_no': '13',
    'x': '126.897336395657',
    'y': '37.4859231683292'},
   'address_name': '서울 구로구 디지털로 306',
   'address_type': 'ROAD_ADDR',
   'road_address': {'address_name': '서울 구로구 디지털로 306',
    'building_name': '대륭포스트타워2차',
    'main_building_no': '306',
    'region_1depth_name': '서울',
    'region_2depth_name': '구로구',
    'region_3depth_name': '구로동',
    'road_name': '디지털로',
    'sub_building_no': '',
    'underground_yn': 'N',
    'x': '126.897336395657',
    'y': '37.4859231683292',
    'zone_no': '08378'},
   'x': '126.897336395657',
   'y': '37.4859231683292'}],
 'meta': {'is_end': True, 'pageable_count': 1, 'total_count': 1}}

In [8]:
resp = response.json()
longitude = resp['documents'][0]['x'] # 경도
latitude = resp['documents'][0]['y'] # 위도
print(longitude, latitude)

126.897336395657 37.4859231683292


### 위도, 경도 찾는 함수 작성

In [14]:
def getLatLng(address):
    app_key = 'KakaoAK ' + '1b131358d811397813db74376a179722'
    url = 'https://dapi.kakao.com/v2/local/search/address.json' # endpoint
    params = {
        'query' : address,
        'page' : '',
        'AddressSize' : ''
    }
    headers = {'Authorization' : app_key}
    response = requests.get(url, headers=headers, params=params)
    status_code = response.status_code
    if status_code == 200:
        resp = response.json()
        longitude = resp['documents'][0]['x'] # 경도
        latitude = resp['documents'][0]['y'] # 위도
        return longitude, latitude
    else:
        return (0, 0)

In [16]:
address = '서울특별시 구로구 디지털로 306'
(x, y) = getLatLng(address)
print('위도 : {}, 경도 : {}'.format(y, x))

위도 : 37.4859231683292, 경도 : 126.897336395657


## 지도에 표시 해보기
- 위도 : 적도를 기준으로 남쪽으로 남극점까지 , 북쪽으로 북국점까지 90도로 나누어 표시
- 경도 : 런던 그리니치천문대를 기준으로 동서로 나누어 동경 180도, 서경 180도 로 분리
- 우리나라 : 적도 북쪽에 있어 북위 34~38, 서울의 경우 동경 127도

### 지도 라이브러리 포리움(Folium)
- pip install folium

In [17]:
pip install folium

In [19]:
import folium

In [22]:
map_thej = folium.Map(location=(y, x), zoom_start=17)
map_thej

In [23]:
map_thej2 = folium.Map(location=(y, x), zoom_start=17, tiles='Stamen Toner')
map_thej2

In [25]:
map_thej3 = folium.Map(location=(y, x), zoom_start=17, tiles='Stamen Terrain')
map_thej3

In [26]:
# 마커 및 팝업 설정
map_thej4 = folium.Map(location=(y, x), zoom_start=17)
folium.Marker((y, x), popup="대륭포스트2차").add_to(map_thej4)
map_thej4

In [34]:
# bootstrap 을 이용해서 마커 설정
# 주피터 노트북이 부트스트랩을 지원 https://icons.getbootstrap.com/  찾아서 넣어볼것
map_thej5 = folium.Map(location=(y, x), zoom_start=17)
folium.Marker((y,x), popup="대륭포스트2차", icon=folium.Icon(color='red', icon='info-sign')).add_to(map_thej5)
map_thej5

In [35]:
map_thej6 = folium.Map(location=(y, x), zoom_start=17)
folium.Marker((y,x), popup="대륭포스트2차", icon=folium.Icon(color='red', icon='info-sign')).add_to(map_thej6)
folium.Circle((y,x), popup="대륭포스트2차", radius=100, color='#3311cc', fill_color='#3311cc').add_to(map_thej6)
map_thej6